aspnet/mobile/native-mobile-backend.rst

```diff
@@ -1,11 +1,170 @@
+
+.. code-block:: csharp
+
+    // URL of REST service (Xamarin ReadOnly Service)
+    //public static string RestUrl = "http://developer.xamarin.com:8081/api/todoitems{0}";
+    
+    // use your machine's IP address
+    public static string RestUrl = "http://192.168.1.207:5000/api/todoitems/{0}";
+
+
+Creating the ASP.NET Core MVC Project
+-------------------------------------
+
+Create a new ASP.NET Core MVC project in Visual Studio. Choose the Web API template and No Authentication.
+
+.. image:: native-mobile-backend/_static/web-api-template.png
+
+The application should respond to all requests made to port 5000. Update `Program.cs` to include ``.UseUrls("http://*:5000")`` to achieve this:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Program.cs
+  :language: c#
+  :lines: 10-16
+  :dedent: 12
+  :emphasize-lines: 3
+
+.. note:: Make sure you run the application using Kestrel, not IIS Express. This can be done from Visual Studio, or by running ``dotnet run`` from the command line.
+
+Add a model class to represent To-Do items. Mark required fields using the ``[Required]`` attribute:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Models/TodoItem.cs
+  :language: c#
+
+The API methods require some way to work with data. Use the same ITodoRepository interface the original Xamarin sample uses:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Interfaces/ITodoRepository.cs
+  :language: c#
+
+For this sample, the implementation just uses a private collection of items. Configure the implementation in `Startup.cs`:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Startup.cs
+  :language: c#
+  :lines: 30-36
+  :dedent: 8
+  :emphasize-lines: 6
+  
+At this point, you're ready to create the `TodoItemsController`.
+
+Creating the Controller
+-----------------------
+
+Add a new controller to the project, `TodoItemsController`. It should inherit from Microsoft.AspNetCore.Mvc.Controller. Add a ``Route`` attribute to indicate that the controller will handle requests made to paths starting with ``api/todoitems``, which may optionally include an ``id`` argument as part of the URL path. The ``[controller]`` token in the route is replaced by the name of the controller (omitting the ``Controller`` suffix), and is especially helpful for global routes. Learn more about :doc:`routing </fundamentals/routing>`.
```

Comment:
Instead of making the id optional on the controller you should instead specify the id parameter on just the specific actions that require it. Otherwise requests like GET api/todoitems/123 will work, even though the id isn't used by the List() method.